INSERT...RETURING statement failed in PostgreSQL DML integration test

## Bug Report

### Which version of ShardingSphere did you use?
master branch, 5.0.0-RC1-SNAPSHOT

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-JDBC

### Expected behavior
Test case pass

### Actual behavior
Exception thrown.
`GeneralDMLIT.assertExecuteUpdate`:
```
/dataset/db/insert_for_order_1.xml, sql=INSERT /*+ index(field1) */ INTO t_order (order_id, user_id, status) VALUES (1, 1, 'insert') RETURNING order_id, ex.msg=传回预期之外的结果。

org.postgresql.util.PSQLException: 传回预期之外的结果。

	at org.postgresql.jdbc.PgStatement.getNoResultUpdateCount(PgStatement.java:254)
	at org.postgresql.jdbc.PgStatement.executeUpdate(PgStatement.java:245)
	at com.zaxxer.hikari.pool.ProxyStatement.executeUpdate(ProxyStatement.java:120)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeUpdate(HikariProxyStatement.java)
	at org.apache.shardingsphere.driver.executor.StatementExecutor$2.executeSQL(StatementExecutor.java:146)
	at org.apache.shardingsphere.driver.executor.StatementExecutor$2.executeSQL(StatementExecutor.java:142)
	at org.apache.shardingsphere.infra.executor.sql.resourced.jdbc.executor.impl.DefaultSQLExecutorCallback.execute0(DefaultSQLExecutorCallback.java:75)
	at org.apache.shardingsphere.infra.executor.sql.resourced.jdbc.executor.impl.DefaultSQLExecutorCallback.execute(DefaultSQLExecutorCallback.java:57)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorKernel.syncExecute(ExecutorKernel.java:99)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorKernel.parallelExecute(ExecutorKernel.java:95)
	at org.apache.shardingsphere.infra.executor.kernel.ExecutorKernel.execute(ExecutorKernel.java:78)
	at org.apache.shardingsphere.infra.executor.sql.resourced.jdbc.executor.SQLExecutor.execute(SQLExecutor.java:66)
	at org.apache.shardingsphere.infra.executor.sql.resourced.jdbc.executor.SQLExecutor.execute(SQLExecutor.java:50)
	at org.apache.shardingsphere.driver.executor.StatementExecutor.executeUpdate(StatementExecutor.java:149)
	at org.apache.shardingsphere.driver.executor.StatementExecutor.executeUpdate(StatementExecutor.java:97)
	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeUpdate(ShardingSphereStatement.java:147)
	at org.apache.shardingsphere.dbtest.engine.dml.GeneralDMLIT.executeUpdateForStatement(GeneralDMLIT.java:80)
	at org.apache.shardingsphere.dbtest.engine.dml.GeneralDMLIT.assertExecuteUpdate(GeneralDMLIT.java:70)
```

`GeneralDMLIT.assertExecute`:
```
INSERT /*+ index(field1) */ INTO t_order (order_id, user_id, status) VALUES (1, 1, 'insert') RETURNING order_id

java.lang.AssertionError: Not a DML statement.

	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.apache.shardingsphere.dbtest.engine.dml.GeneralDMLIT.executeForStatement(GeneralDMLIT.java:115)
	at org.apache.shardingsphere.dbtest.engine.dml.GeneralDMLIT.assertExecute(GeneralDMLIT.java:105)
```

### Reason analyze (If you can)
INSERT...RETURNING need special code to operate.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
1, env.properties:
rule.types=db
databases=PostgreSQL
postgresql.host=127.0.0.1
postgresql.password=test

2, Run GeneralDMLIT in IDEA

### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/cases/assertion/dml/DMLIntegrateTestCaseAssertion.java
+++ b/shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/cases/assertion/dml/DMLIntegrateTestCaseAssertion.java
@@ -17,6 +17,7 @@
 
 package org.apache.shardingsphere.dbtest.cases.assertion.dml;
 
+import javax.xml.bind.annotation.XmlAttribute;
 import lombok.Getter;
 import org.apache.shardingsphere.dbtest.cases.assertion.root.IntegrateTestCaseAssertion;
 
@@ -25,4 +26,7 @@ import org.apache.shardingsphere.dbtest.cases.assertion.root.IntegrateTestCaseAs
  */
 @Getter
 public final class DMLIntegrateTestCaseAssertion extends IntegrateTestCaseAssertion {
+    
+    @XmlAttribute(name = "dml-returning")
+    private boolean dmlReturning;
 }

--- a/shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/engine/dml/GeneralDMLIT.java
+++ b/shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/engine/dml/GeneralDMLIT.java
@@ -17,6 +17,7 @@
 
 package org.apache.shardingsphere.dbtest.engine.dml;
 
+import java.sql.ResultSet;
 import org.apache.shardingsphere.dbtest.cases.assertion.dml.DMLIntegrateTestCaseAssertion;
 import org.apache.shardingsphere.dbtest.cases.assertion.root.SQLValue;
 import org.apache.shardingsphere.dbtest.cases.assertion.root.SQLCaseType;
@@ -63,6 +64,9 @@ public final class GeneralDMLIT extends BaseDMLIT {
         }
         // TODO fix shadow
         if ("shadow".equals(getRuleType())) {
+            return;
+        }
+        if (assertion.isDmlReturning()) {
             return;
         }
         int actualUpdateCount;
@@ -112,6 +116,15 @@ public final class GeneralDMLIT extends BaseDMLIT {
     
     private int executeForStatement(final Connection connection) throws SQLException {
         try (Statement statement = connection.createStatement()) {
+            if (assertion.isDmlReturning()) {
+                if (statement.execute(getSql())) {
+                    ResultSet resultSet = statement.getResultSet();
+                    boolean moreResults = statement.getMoreResults();
+                    // TODO resultSet is null
+                    return moreResults || null != resultSet ? statement.getUpdateCount() : 1;
+                }
+                return statement.getUpdateCount();
+            }
             assertFalse("Not a DML statement.", statement.execute(getSql()));
             return statement.getUpdateCount();
         }
@@ -121,6 +134,15 @@ public final class GeneralDMLIT extends BaseDMLIT {
         try (PreparedStatement preparedStatement = connection.prepareStatement(getSql())) {
             for (SQLValue each : assertion.getSQLValues()) {
                 preparedStatement.setObject(each.getIndex(), each.getValue());
+            }
+            if (assertion.isDmlReturning()) {
+                if (preparedStatement.execute()) {
+                    ResultSet resultSet = preparedStatement.getResultSet();
+                    boolean moreResults = preparedStatement.getMoreResults();
+                    // TODO resultSet is null
+                    return moreResults || null != resultSet ? preparedStatement.getUpdateCount() : 1;
+                }
+                return preparedStatement.getUpdateCount();
             }
             assertFalse("Not a DML statement.", preparedStatement.execute());
             return preparedStatement.getUpdateCount();

--- a/shardingsphere-integration-test/shardingsphere-test-suite/src/test/resources/integrate/cases/dml/dml-integrate-test-cases.xml
+++ b/shardingsphere-integration-test/shardingsphere-test-suite/src/test/resources/integrate/cases/dml/dml-integrate-test-cases.xml
@@ -34,7 +34,7 @@
     </dml-test-case>
     
     <dml-test-case sql="INSERT /*+ index(field1) */ INTO t_order (order_id, user_id, status) VALUES (1, 1, 'insert') RETURNING order_id" db-types="PostgreSQL">
-        <assertion expected-data-file="insert_for_order_1.xml" />
+        <assertion expected-data-file="insert_for_order_1.xml" dml-returning="true" />
     </dml-test-case>
     
     <dml-test-case sql="INSERT INTO t_order (order_id, user_id, status) VALUES (1, 1, 'insert'), (2, 2, 'insert2')" db-types="MySQL">

